app/dashboard/views.py

```diff
@@ -731,6 +731,50 @@ def onboard(request, flow):
     return TemplateResponse(request, 'ftux/onboard.html', params)
 
 
+def users_directory(request):
+    """Handle displaying users directory page."""
+
+    params = {
+        'active': 'users',
+        'title': 'Users',
+        'meta_title': "",
+        'meta_description': ""
+    }
+    return TemplateResponse(request, 'dashboard/users.html', params)
+
+
+@require_GET
+def users_fetch(request):
+    """Handle displaying users."""
+    q = request.GET.get('search','')
+    limit = int(request.GET.get('limit', 10))
+    page = int(request.GET.get('page', 1))
+    order_by = request.GET.get('order_by', '-created_on')
+    context = {}
+    user_list = Profile.objects.all().order_by(order_by).filter(handle__icontains=q).cache()
+
+    # all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('-id')
+    params = dict()
+    all_pages = Paginator(user_list, limit)
+    all_users = []
+    for user in all_pages.page(page):
+        print(user)
+        profile_json = {}
+        profile_json = user.to_standard_dict()
+        if user.avatar_baseavatar_related.exists():
+            profile_json['avatar_id'] = user.avatar_baseavatar_related.first().pk
```

Comment:
Better here to assign a var to `user.avatar_baseavatar_related.first()` and then access the `.pk` and `.avatar_url` from that var. Calling `.first()` multiple times re-queries the database